hnn_core/optimization.py

```diff
@@ -430,12 +529,27 @@ def optimize_evoked(params, target_dpl, initial_dpl, maxiter=50,
         print("Starting optimization step %d/%d" % (step + 1, total_steps))
 
         opt_params['optiter'] = 0
-        opt_params['stepminopterr'] = 1e9  # min optimization error so far
-        opt_dpls = dict(best_dpl=None, target_dpl=target_dpl)
-
-        def _myoptrun(new_params):
-            return _optrun(new_params, opt_params,
-                           params, opt_dpls=opt_dpls,
+        opt_params['stepminopterr'] = _rmse(opt_dpls['best_dpl'],
+                                            opt_dpls['target_dpl'],
+                                            tstart=opt_params['opt_start'],
+                                            tstop=opt_params['opt_end'],
+                                            weights=opt_params['weights'])
+
+        net_opt = net.copy()
+        # XXX use functools.partial instead?
+        # updated_drive_params must be a list for compatability with the args
```

Comment:
```suggestion
        # drive_params_updated must be a list for compatability with the args
```

I would put this comment when you instantiate `drive_params_updated` though. As in:

```py
drive_params_updated = list()  # for compatibility with args in scipy.optimize.fmin.cobyla
```

Reply:
That's the thing, the variable `drive_params_updated` is never explicitly defined *as a list* within any of the hnn-core namespaces, but is rather passed into the `_optrun` callable via `fmin_cobyla`. It's really confusing, so I placed the comment at the first place in the code that mentions this abstract variable (i.e., near the definition of the `_myoptrun` namespace) that might help us understand why it's important to place the drive parameters that undergo optimization in an *ordered* data structure.

I'm sure there's a better way to clarify this in the codebase, but have yet to arrive at a good solution. Blake did a great job with this part of the code but it's taken me awhile to wrap my head around. Any suggestions?